server/dbInit.test.ts

```diff
@@ -0,0 +1,157 @@
+// Set up fake endpoints to query
+import express from "express";
+import axios from "axios";
+import { MongoMemoryServer } from "mongodb-memory-server";
+import mongoose from "mongoose";
+import { Subjects, Classes } from "./dbDefs";
+import { fetchSubjects, fetchClassesForSubject, fetchAddCourses } from "./dbInit";
+
+// May require additional time for downloading 100 mb (!) worth of MongoDB binaries
+// **We might not want to run this with CI**
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;
+
+let testServer: MongoMemoryServer;
+let serverCloseHandle;
+
+const testingPort = 27760;
+const testingEndpoint = `http://localhost:${testingPort}/`;
+
+// Configure a mongo server and fake endpoints for the tests to use
+beforeAll(async () => {
+  // get mongoose all set up
+  testServer = new MongoMemoryServer();
+  const mongoUri = await testServer.getUri();
+  await mongoose.connect(mongoUri, { useNewUrlParser: true, useUnifiedTopology: true });
+
+  await new Subjects({
+    _id: "some id",
+    subShort: "GORK",
+    subFull: "Study of Angry Fungi",
+  }).save();
+
+  await new Classes({
+    _id: "some other id",
+    classSub: "GORK",
+    classNum: "1110",
+    classTitle: "Introduction to Angry Fungi",
+    classFull: "GORK 1110 Introduction to Angry Fungi",
+    classSems: ["FA19"],
+    classProfessors: ["Prof. Thraka"],
+    classRating: 5,
+    classWorkload: 5,
+    classDifficulty: 5,
+  }).save().catch((err) => { console.log(err); });
```

Comment:
Sorry should have clarified this before: for the sake of clarity and usefulness in the logs, could you please log an explanation of the log error maybe like this to all these logs?

https://github.com/cornell-dti/course-reviews-react-2.0/blob/f30c4ca2f22682e574e47f703c3fd1100d558f6b/server/methods.ts#L198

Reply:
If this fails, then tests start failing. In particular, one which checks if GORK 1110 is in the db. See line 106.

Reply:
Makes sense